File: register_github.py
#!/usr/bin/env python3
"""
使用 Playwright 注册 GitHub 账号
"""

import asyncio
from playwright.async_api import async_playwright

async def register_github():
    async with async_playwright() as p:
        # 启动浏览器
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context()
        page = await context.new_page()
        
        print("🌐 访问 GitHub 注册页面...")
        await page.goto("https://github.com/signup")
        
        # 等待页面加载
        await page.wait_for_timeout(3000)
        
        # 截图看看当前状态
        await page.screenshot(path="/tmp/github_signup.png")
        print("📸 截图保存到 /tmp/github_signup.png")
        
        # 获取页面内容
        content = await page.content()
        print("📄 页面标题:", await page.title())
        
        await browser.close()
        print("✅ 浏览器已关闭")

if __name__ == "__main__":
    asyncio.run(register_github())

File: register_github.js
const puppeteer = require('puppeteer');

async function registerGitHub() {
    console.log('🚀 启动浏览器...');
    
    const browser = await puppeteer.launch({
        headless: 'new',
        executablePath: '/home/node/.cache/ms-playwright/chromium-1208/chrome-linux64/chrome',
        args: [
            '--no-sandbox',
            '--disable-setuid-sandbox',
            '--disable-dev-shm-usage',
            '--disable-gpu'
        ]
    });
    
    const page = await browser.newPage();
    
    console.log('🌐 访问 GitHub 注册页面...');
    await page.goto('https://github.com/signup', { waitUntil: 'networkidle2' });
    
    // 截图
    await page.screenshot({ path: '/tmp/github_signup.png', fullPage: true });
    console.log('📸 截图保存到 /tmp/github_signup.png');
    
    // 获取页面信息
    const title = await page.title();
    console.log('📄 页面标题:', title);
    
    // 获取表单字段
    const inputs = await page.$$eval('input', inputs => 
        inputs.map(i => ({ name: i.name, id: i.id, type: i.type, placeholder: i.placeholder }))
    );
    console.log('📝 表单字段:', JSON.stringify(inputs.slice(0, 10), null, 2));
    
    // 获取页面文本
    const bodyText = await page.$eval('body', el => el.innerText);
    console.log('📄 页面内容预览:', bodyText.substring(0, 500));
    
    await browser.close();
    console.log('✅ 完成');
}

registerGitHub().catch(console.error);
